### PR TITLE
NASM Support

### PR DESCRIPTION
**Pull request by gingerBill**
Allow `.asm` files to be compiled with NASM through `foreign import`

### CODE REVIEW

**Review by Kelimion — COMMENTED** [review]
Looks good to me on the whole.

File: core/runtime/procs_windows_amd64.odin [hunk 1]
```diff
@@ -20,13 +20,6 @@ windows_trap_type_assertion :: proc "contextless" () -> ! {
 }
 
 when ODIN_NO_CRT {
```
[the remark this answers is not in the document]
Comment (gingerBill) [reply core/runtime/procs_windows_amd64.odin:22]: Note the file is only for windows and amd64

File: src/main.cpp [hunk 2]
```diff
@@ -212,28 +211,43 @@ i32 linker_stage(lbGenerator *gen) {
...
 
 		for_array(j, gen->modules.entries) {
 			lbModule *m = gen->modules.entries[j].value;
 			for_array(i, m->foreign_library_paths) {
 				String lib = m->foreign_library_paths[i];
-				GB_ASSERT(lib.len < gb_count_of(lib_str_buf)-1);
-				gb_snprintf(lib_str_buf, gb_size_of(lib_str_buf),
-							" \"%.*s\"", LIT(lib));
-				lib_str = gb_string_appendc(lib_str, lib_str_buf);
+				if (string_ends_with(lib, str_lit(".asm"))) {
```
[the remark this answers is not in the document]
Comment (gingerBill) [reply src/main.cpp:228]: I could support .s and .S too, good point.

[the remark this answers is not in the document]
Comment (gingerBill) [reply src/main.cpp:228]: No need for any directives, need the file extension.

**Review by Kelimion — APPROVED** [review]
So far, so good.

**Pull request merged**